File: common_book/src/mfxsqbyd/java/com/intelligent/reader/search/RecommendBooksAdapter.java
package com.intelligent.reader.search;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.ding.basic.bean.SearchRecommendBook;
import com.intelligent.reader.R;


import java.lang.ref.WeakReference;
import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.List;

/**
 * Function：搜索页-推荐书籍子条目
 *
 * Created by JoannChen on 2018/5/30 0030 17:28
 * E-mail:yongzuo_chen@dingyuegroup.cn
 */
public class RecommendBooksAdapter extends RecyclerView.Adapter<RecommendBooksAdapter.ViewHolder> {

    private WeakReference<Context> weakReference;
    private RecommendItemClickListener recommendItemClickListener;
    private List<SearchRecommendBook.DataBean> books = new ArrayList<>();

    RecommendBooksAdapter(Context context,
            RecommendItemClickListener recommendItemClickListener,
            List<SearchRecommendBook.DataBean> books) {
        this.weakReference = new WeakReference<>(context);
        this.recommendItemClickListener = recommendItemClickListener;
        this.books = books;

    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater.from(weakReference.get()).inflate(R.layout.item_search_recommend,
                        parent, false), recommendItemClickListener);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        SearchRecommendBook.DataBean book = books.get(position);
        holder.tv_book_name.setText(book.getBookName());
        holder.tv_author.setText((book.getAuthorName()));
        if (holder.iv_url != null && !TextUtils.isEmpty(book.getSourceImageUrl())) {
            Glide.with(weakReference.get()).load(book.getSourceImageUrl()).placeholder(
                    net.lzbook.kit.R.drawable.icon_book_cover_default)
                    .error((net.lzbook.kit.R.drawable.icon_book_cover_default))
                    .into(holder.iv_url);
        } else {
            Glide.with(weakReference.get()).load(
                    net.lzbook.kit.R.drawable.icon_book_cover_default).into(holder.iv_url);
        }


    }

    @Override
    public int getItemCount() {
        if (books.size() != 0) {
            return books.size();
        }
        return 0;
    }

    public interface RecommendItemClickListener {
        void onItemClick(View view, int position);
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private RecommendItemClickListener recommendItemClickListener;
        private ImageView iv_url;
        private TextView tv_book_name, tv_author;

        public ViewHolder(View itemView, RecommendItemClickListener recommendItemClickListener) {
            super(itemView);
            this.recommendItemClickListener = recommendItemClickListener;
            itemView.setOnClickListener(this);
            iv_url = (ImageView) itemView.findViewById(R.id.iv_url);
            tv_book_name = (TextView) itemView.findViewById(R.id.tv_book_name);
            tv_author = (TextView) itemView.findViewById(R.id.tv_author);
        }

        @Override
        public void onClick(View v) {

            if (recommendItemClickListener != null) {
                recommendItemClickListener.onItemClick(v, getPosition());
            }
        }
    }

}

File: common_book/src/qbmfkkydq/java/com/intelligent/reader/adapter/RecommendBooksAdapter.kt
package com.intelligent.reader.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.ding.basic.bean.SearchRecommendBook
import com.intelligent.reader.R

import java.lang.ref.WeakReference

/**
 * Function：搜索推荐书籍
 *
 * Created by JoannChen on 2018/7/19 0019 16:28
 * E-mail:yongzuo_chen@dingyuegroup.cn
 */
class RecommendBooksAdapter(context: Context,
                            private val recommendItemClickListener: RecommendItemClickListener,
                            val books: List<SearchRecommendBook.DataBean>) : RecyclerView.Adapter<RecommendBooksAdapter.ViewHolder>() {

    private val weakReference: WeakReference<Context> = WeakReference(context)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
                LayoutInflater.from(weakReference.get()).inflate(R.layout.item_search_recommend,
                        parent, false), recommendItemClickListener)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val book = books[position]
        holder.bookNameText.text = book.bookName
        holder.authorText.text = book.authorName
        if (!TextUtils.isEmpty(book.sourceImageUrl)) {
            Glide.with(weakReference.get()).load(book.sourceImageUrl).placeholder(
                    net.lzbook.kit.R.drawable.icon_book_cover_default)
                    .error(net.lzbook.kit.R.drawable.icon_book_cover_default)
                    .into(holder.urlImageView)
        } else {
            Glide.with(weakReference.get()).load(
                    net.lzbook.kit.R.drawable.icon_book_cover_default).into(holder.urlImageView)
        }

    }

    override fun getItemCount(): Int {
        return books.size
    }

    interface RecommendItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    inner class ViewHolder(itemView: View, private val recommendItemClickListener: RecommendItemClickListener?) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val urlImageView: ImageView
        val bookNameText: TextView
        val authorText: TextView

        init {
            itemView.setOnClickListener(this)
            urlImageView = itemView.findViewById(R.id.iv_url)
            bookNameText = itemView.findViewById(R.id.tv_book_name)
            authorText = itemView.findViewById(R.id.tv_author)

        }

        override fun onClick(v: View) {

            recommendItemClickListener?.onItemClick(v, position)
        }
    }

}
